src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -93,32 +110,56 @@ string_exprt string_constraint_generatort::add_axioms_from_bool(
 /// Add axioms enforcing that the string corresponds to the result
 /// of String.valueOf(I) or String.valueOf(J) Java functions applied
 /// on the integer expression.
-/// \param i: a signed integer expression
+/// \param x: a signed integer expression
 /// \param max_size: a maximal size for the string representation
+/// \param radix: an expression for the radix
 /// \param ref_type: type for refined strings
 /// \return a string expression
 string_exprt string_constraint_generatort::add_axioms_from_int(
-  const exprt &i, size_t max_size, const refined_string_typet &ref_type)
+  const exprt &x,
+  const exprt radix,
+  const refined_string_typet &ref_type)
 {
-  PRECONDITION(i.type().id()==ID_signedbv);
+  /// If we cannot tell what the radix is then so we assume it is 2 to make sure
+  /// max_size will definitely be large enough
+  double radix_d=static_cast<double>(try_to_evaluate_expr_as_ul(radix, 2ul));
+  size_t max_size=calculate_max_string_length(x.type(), radix_d);
```

Comment:
1. `calculate_max_string_length` -> `max_printed_integer_length`
2. Take `radix` as an integer and cast to double within that function, since passing a fractional value is not sensible